### PR TITLE
Implement custom join/leave messages

### PR DESCRIPTION
**Pull request by AndreasHGK**
This PR adds a basic implementation of customizable join and leave messages.

You can set the messages both in the `config.toml` file or directly with code for each player.

### CODE REVIEW

File: dragonfly/session/text.go [hunk 1]
```diff
@@ -134,3 +134,13 @@ func (s *Session) SendSubtitle(text string) {
 func (s *Session) SendActionBarMessage(text string) {
 	s.writePacket(&packet.SetTitle{ActionType: packet.TitleActionSetActionBar, Text: text})
 }
+
+// SetJoinMessage changes the message that appears when the player joins the server
+func (s *Session) SetJoinMessage(message string) {
+	s.joinMessage = message
```
Comment (Sandertv) [inline dragonfly/session/text.go:140]: I think these properties are vulnerable to race conditions. Consider changing them to an `atomic.String`

File: dragonfly/player/player.go [hunk 2]
```diff
@@ -256,6 +256,16 @@ func (p *Player) RemoveBossBar() {
...
 
+// SetJoinMessage changes the message that appears when the player joins the server
+func (p *Player) SetJoinMessage(message string) {
+	p.s.SetJoinMessage(message)
+}
+
+// SetLeaveMessage changes the message that appears when the player leaves the server
+func (p *Player) SetLeaveMessage(message string) {
+	p.s.SetLeaveMessage(message)
+}
```
Comment (Sandertv) [inline dragonfly/player/player.go:267]: I'm not too fond of these messages and the one in Session. Maybe just SetJoinMessage and SetLeaveMessage in server.go would be better?

File: dragonfly/config.go [hunk 3]
```diff
@@ -20,6 +20,12 @@ type Config struct {
 		ShutdownMessage string
 		// AuthEnabled controls whether or not players must be connected to Xbox Live in order to join the server.
 		AuthEnabled bool
+		// JoinMessage is the message that appears when a player joins the server. Leave this empty to disable it.
+		// %v is the placeholder for the username of the player
```
Comment (Sandertv) [inline dragonfly/config.go:24]: Make sure to add this documentation in the respective methods too.

**Review by Sandertv — CHANGES_REQUESTED** [review]
Thanks for you PR! I've left some comments on the diff.

File: dragonfly/server.go [hunk 4]
```diff
@@ -217,6 +221,30 @@ func (server *Server) SetName(a ...interface{}) {
 	server.name.Store(fmt.Sprint(a...))
 }
 
+// JoinMessage is the message that appears when a player joins the server. Leave this empty to disable it.
+// %v is the placeholder for the username of the player
+func (server *Server) JoinMessage() string {
```
Comment (Sandertv) [inline dragonfly/server.go:226]: Is there actually any use for this?

File: dragonfly/server.go [hunk 5]
```diff
@@ -217,6 +221,30 @@ func (server *Server) SetName(a ...interface{}) {
...
+
+// SetJoinMessage changes the join message for all players on the server. Leave this empty to disable it.
+//// %v is the placeholder for the username of the player
+func (server *Server) SetJoinMessage(message string) {
+	server.joinMessage.Store(message)
+}
+
+// LeaveMessage is the message that appears when a player leaves the server. Leave this empty to disable it.
+// %v is the placeholder for the username of the player
+func (server *Server) LeaveMessage() string {
```
Comment (Sandertv) [inline dragonfly/server.go:238]: Same as above.

**Review by Sandertv — COMMENTED** [review]
Two comments, looks pretty good now!

File: dragonfly/server.go (hunk 4 above)
Comment (AndreasHGK) [reply dragonfly/server.go:226]: The use is just to return what the current join/leave message is

Comment (Sandertv) [reply dragonfly/server.go:226]: Yeah, but is there any reason anyone would like... Have any use for that?

Comment (AndreasHGK) [reply dragonfly/server.go:226]: I don't know lol, I just added it in case someone somewhere would need it.

**Review by Sandertv — APPROVED** [review]
Looks good, thank you for the PR!

**Pull request merged**